advancedblacklist/commands/blacklist.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (c) 2021 - Jojo#7791
+# Licensed under MIT
+
+import discord
+from redbot.core import commands
+from redbot.core.utils.chat_formatting import pagify
+
+from ..abc import ABMixin  # type:ignore
+from .utils import (add_to_blacklist, edit_reason, get_blacklist,
+                    in_blacklist, remove_from_blacklist)
+
+__all__ = ["Blacklist"]
+
+
+class Blacklist(ABMixin):
+    """Commands for the blacklist side of the cog"""
+
+    @commands.group(name="blacklist", aliases=["blocklist"])
+    @commands.is_owner()
+    async def blacklist(self, ctx: commands.Context):
+        """Manage [botname]'s blacklist"""
+        pass
+
+    @blacklist.command(name="add")
+    async def blacklist_add(
+        self, ctx: commands.Context, users: commands.Greedy[discord.User], *, reason: str = None
+    ):
+        """Add users to the blacklist.
+
+        **Arguments**
+            - `users` The users to add to the blacklist. These cannot be bots.
+            - `reason` The reason for adding these users to the blacklist. This is optional.
+        """
+        if not users:
+            raise commands.UserInputError
+        for user in users:
+            if user.bot:
+                return await ctx.send("You cannot add bots to the blacklist.")
+            elif await self.bot.is_owner(user):
+                return await ctx.send("You cannot add a bot owner to the blacklist.")
+
+        reason = reason or "No reason provided."
+        await add_to_blacklist(self.bot, users, reason)
+        that = "that user" if len(users) == 1 else "those users"
+        await ctx.send(f"Done. Added {that} to the blacklist.")
+
+    @blacklist.command(name="remove", aliases=["del", "delete", "rm"], require_var_positional=True)
+    async def blacklist_remove(self, ctx: commands.Context, *users: discord.User):
+        """Remove users from the blacklist.
+
+        **Arguments**
+            - `users` The users to remove from the blacklist.
+        """
+        if not await get_blacklist(self.bot):
+            return await ctx.send("There are no users in the blacklist.")
+        await remove_from_blacklist(self.bot, users)
+        that = "that user" if len(users) == 1 else "those users"
+        await ctx.send(f"Done. Removed {that} from the blacklist.")
+
+    @blacklist.command(name="reason")
+    async def blacklist_reason(self, ctx: commands.Context, user: discord.Member, *, reason: str):
```

Comment:
```suggestion
    async def blacklist_reason(self, ctx: commands.Context, user: discord.User, *, reason: str):
```
Seeing as this is for the global blacklist and not the local counterpart, shouldn't this also be type hinted as discord.User and *not* discord.Member?